### DISCUSSION TITLE
Valetudo 2021.09.0

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1075] [Release Discussions]
<div align="center">
    <img src="https://github.com/Hypfer/Valetudo/blob/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2021.09.0</h2></p>
</div>

This release features some more bugfixes and some breaking config changes

## New UI

If you haven't seen it already, check out [the previous release notes](https://github.com/Hypfer/Valetudo/releases/tag/2021.08.1).

### Config Schema
This release features a much cleaner mqtt configuration schema. Now, one can actually understand what the options are for.

![image](https://user-images.githubusercontent.com/974410/132738325-10a47c75-f321-4c79-94ec-374484d7b7f4.png)

Since this a **breaking change**, Valetudo will likely reject your old config file and create a new one.
Don't worry though. The old one is backed up meaning that you can simply copy-paste your timers, presets etc.
Just check the Log for more information.

### Identifier

Valetudo will now use the autogenerated machine identifier as the mqtt identifier and friendly name.
If you've been using the defaults until now, you'll either want to manually configure the identifier or update your scripts, delete your Vacuum Device in Home Assistant etc.

The autogenerated unique machine identifier can be found in the Log.
It will sorta look like this: `ModestFewChinchilla`

### Home Assistant Map Data

The Map data camera image has been replaced with a nicer looking one that will also hopefully lead to less confusion for newcomers.
It looks like this:

![image](https://user-images.githubusercontent.com/974410/132737756-efba2973-0ed7-47cb-bae4-6b6ca18acf67.png)

## Out-of-Memory Issues

The out of memory issues causing Valetudo to shut down have been fixed again even though that they were already fixed.
For some reason, npm started installing the old and unpatched dependency. Probably some weirdness regarding the `package-lock.json` or something like that.

This has been solved by reimplementing what the dependency did in Valetudo itself. Fortunately, this allowed us to add logging to that.
If you're seeing something like `[WARN] Stale SSE connection to the Map SSE Hub detected. Terminating.` in your log, the mitigation is working.

## Updated Robot Docs

I've added some notes for each implementation to the [Supported Robots Page](https://valetudo.cloud/pages/general/supported-robots.html) to make it easier for people to choose a robot to buy.

### Features

* Log autogenerated system ID ([2cc365c](https://github.com/Hypfer/Valetudo/commit/2cc365c60415f68e3ec7d454eda8387adf8c9842))
* **ui:** make toggle group full width ([#1060](https://github.com/Hypfer/Valetudo/issues/1060)) ([dfdd7f7](https://github.com/Hypfer/Valetudo/commit/dfdd7f7de54ad84f0f4fd52e34692879ce0116ff))

### Bug Fixes

* **miio:** reduce loglevel of user ack timeout spam messages ([3f81d4c](https://github.com/Hypfer/Valetudo/commit/3f81d4c063595f129d7b08114377b6b4703628e1))
* **mqtt:** Poll robot state every refreshInterval to mitigate stale data on lost cloud connections ([95ce4bb](https://github.com/Hypfer/Valetudo/commit/95ce4bb333911f67afadd769af4104d8286dae6b))
* **openapi:** Fix examples schema for ManualControlCapability ([b22ceb7](https://github.com/Hypfer/Valetudo/commit/b22ceb7288f87540492182eba58f821fe44a544e))
* **ui:** Change path color to black for better visibility ([2454a85](https://github.com/Hypfer/Valetudo/commit/2454a85a650a9323293dd9190cf5a955b8b2e7f8))
* **vendor.viomi:** catch set_timezone errors ([c2cbd45](https://github.com/Hypfer/Valetudo/commit/c2cbd4588dede2b4d6a86c7d0807b16d7308d0f2))

### Reverts

* Revert "fix(vendor.viomi): Poll state every 30s to fix stale mqtt data" ([b55e4a5](https://github.com/Hypfer/Valetudo/commit/b55e4a58edc20d3d7e3f67274891d46967f0fd8d))

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2021.09.0'>Valetudo 2021.09.0</a>.</em>

**Comment by StefanSchoof** [comment]
I have tested the new UI. Is working smooth for me.
I often use the segments to clean only my kitchen or other areas in my house. In the new UI this feature is further away and since I am lazy I switched back to use the old UI.
